### PR TITLE
Use dependabot and dropping Python3.6 support

### PR DESCRIPTION
**Pull request by tkoyama010**
Using depandabot: https://github.com/pyvista/pyvista/pull/1972 .
Dropping Python3.6 and Python3 support: https://github.com/pyvista/pyvista/pull/1716

To solve the CI error in https://github.com/pyvista/pyvistaqt/pull/161 .

### CODE REVIEW

File: requirements_test.txt [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+pydocstyle==6.1.1
+pylint==2.13.0
+pytest==7.1.2
+pytest-cov==3.0.0
+pytest-memprof==0.2.0
+pytest-qt==4.0.2
+pyvista==0.34.1
+QtPy==2.1.0
+scooby==0.5.12
+vtk==9.0.1
```
Comment (tkoyama010) [inline requirements_test.txt:19]:
```suggestion
```

**Comment by tkoyama010** [comment]
In the future, we need to do https://github.com/pyvista/pyvista/pull/2083. But that will be another PR.

File: requirements_test.txt [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
-pycodestyle
-mypy
-flake8
-codespell
-pydocstyle
+black==22.3.0
+codecov==2.1.12
+codespell==2.1.0
+flake8==4.0.1
+ipython==8.2.0
```
Comment (tkoyama010) [inline requirements_test.txt:5]:
```suggestion
ipython==7.34.0
```

File: requirements_test.txt [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
-codespell
-pydocstyle
+black==22.3.0
+codecov==2.1.12
+codespell==2.1.0
+flake8==4.0.1
+ipython==7.34.0
+isort==5.10.1
+mypy==0.950
+numpy==1.22.3
```
Comment (tkoyama010) [inline requirements_test.txt:8]:
```suggestion
numpy==1.21.6
```

**Review by akaszynski — APPROVED** [review]

**Comment by akaszynski** [comment]
Didn't see this one. Thanks @tkoyama010!

**Pull request merged**